src/v/cluster/tx_gateway_frontend.cc

```diff
@@ -839,14 +886,26 @@ ss::future<cluster::init_tm_tx_reply> tx_gateway_frontend::init_tm_tx_locally(
       tx_id,
       transaction_timeout_ms);
 
-    auto shard = _shard_table.local().shard_for(model::tx_manager_ntp);
+    auto tx_ntp_opt = _tx_coordinator_ntp_mapper.local().ntp_for(
```

Comment:
make init_tm_tx_locally private, pass partition_id from init_tm_tx, invoke init_tm_tx instead of init_tm_tx_locally from tx_gateway